### PR TITLE
Add possibility to disable configmap creation

### PR DESCRIPTION
**Pull request by ugwun**
As configmaps are sometimes managed by different parties (in our case it is a Helmchart that takes care of creating and destroying configmaps), it would be helpful to have a possibility to disable leader configmap creation on the library's side.

### CODE REVIEW

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]
Can you provide a test?

**Comment by ugwun** [comment]
I added unit test org.springframework.cloud.kubernetes.fabric8.leader.Fabric8LeadershipControllerTest#whenNonExistentConfigmapAndCreationNotAllowedStopLeadershipAcquire

**Comment by ryanjbaxter** [comment]
Thanks, it would be good to add some documentation about the property as well.

File: spring-cloud-kubernetes-fabric8-leader/src/test/java/org/springframework/cloud/kubernetes/fabric8/leader/Fabric8LeadershipControllerTest.java [hunk 1]
```diff
@@ -60,4 +74,55 @@ public void shouldGetEmptyLocalLeader() {
 		assertThat(this.fabric8LeadershipController.getLocalLeader().isPresent()).isFalse();
 	}
 
+	@Test
+	void whenNonExistentConfigmapAndCreationNotAllowedStopLeadershipAcquire() {
+
+		class MemoryAppender extends ListAppender<ILoggingEvent> {
```
Comment (wind57) [inline spring-cloud-kubernetes-fabric8-leader/src/test/java/org/springframework/cloud/kubernetes/fabric8/leader/Fabric8LeadershipControllerTest.java:80]: so this is supposed to capture a log and assert it, right? If so, can't we use `OutputCaptureExtension' here? wdyt?

Comment (ugwun) [reply spring-cloud-kubernetes-fabric8-leader/src/test/java/org/springframework/cloud/kubernetes/fabric8/leader/Fabric8LeadershipControllerTest.java]: I didn't know this possibility existed. I updated the code with OutputCaptureExtension. Thanks.

**Comment by ugwun** [comment]
@ryanjbaxter I added documentation to `_configprops.adoc`

**Pull request merged**